identity-wallet/src/state/user_data_query/mod.rs

```diff
@@ -1,2 +1,19 @@
 pub mod actions;
 pub mod reducers;
+
+use super::FeatTrait;
+
+use derivative::Derivative;
+use serde::{Deserialize, Serialize};
+use ts_rs::TS;
+
+#[derive(Default, Serialize, Deserialize, Derivative, TS, Clone, PartialEq, Debug)]
+#[ts(export)]
+#[serde(default)]
+pub struct UserDataQueryResults {
```

Comment:
Wouldn't it be sufficient to just call it `UserDataQuery` since you're having the `results` inside anyway? Otherwise you get stuff like this: `user_data_query_results.results`. I think `user_data_query.results` should be good, no?

Reply:
So my thought is this:
the name UserDataQuery is already used, its the action dispatched by the frontend and the accompanying payload.
I don't necessarily like the name UserDataQueryResults too much either, but it makes sense to me. The user queries from the FE, the BE sends the Results back. to prevent the ~results.results situation I can rename the results field to query or indices? Anyway I'm open to any suggestion, but i do agree with you, it shouldn't be ~results.results

Reply:
Like you mentioned, perhaps now is a good time to change the `UserDataQuery` Action to something more 'Search' related (following these recommendations: https://redux.js.org/style-guide/#model-actions-as-events-not-setters). 
Options could be one of these?:
>* SearchPerformed
>* SearchInitiated
>* SearchStarted
>* SearchSubmitted
>* SearchRequested

Then `UserDataQueryResults` could be renamed to `SearchResults` for example.

Reply:
Haha I still lean towards option 6: "SearchQuery" if you don't mind :)
SearchQuery & SearchResults